Simplify code for JSON audit log

JakubOnderka opened the pull request:
`yajl_gen_string` doesn't require null terminated string, so we don't need to call `strlen`.

This simplifies code for JSON transaction logging.

JakubOnderka committed:
Simplify code for JSON audit log

airween commented:
Hi @JakubOnderka,

thanks for this PR!

Have you seen the issue #3310? There the OP mentioned a [commit](https://github.com/paul-hammant/ModSecurity/commit/7010ddd8222afa2608ce506697d7c20187364a84) - may be we can consider to add that modification too. What do you think?

JakubOnderka committed:
Test for JSON audit log

JakubOnderka commented:
Good question. So I added test case that verifies if timestamp is correct in JSON format and it looks good.

JakubOnderka committed:
Change http_version field in JSON audit log to string

Sometimes m_httpVersion variable can be empty and then invalid JSON is generated

JakubOnderka commented:
@airween I also find out that in some cases, JSON log is invalid because `m_httpVersion` variable is empty string. So I converted `http_version` field in JSON audit log to string, so if  `m_httpVersion` will be empty, JSON will still be valid.

airween reviewed (approved):
LGTM

airween commented:
Hi @JakubOnderka,

a nice catch again - thank you!
